plugin-server/tests/main/ingestion-queues/analytics-events-ingestion-consumer.test.ts

```diff
@@ -47,58 +51,59 @@ const captureEndpointEvent2 = {
+        return event.token + ':' + event.distinct_id
+    }
+
+    function createBatchWithMultipleEvents(events: any[], timestamp?: any, withKey: boolean = true): Message[] {
+        return events.map((event, i) => ({
             partition: 0,
             topic: KAFKA_EVENTS_PLUGIN_INGESTION,
-            value: JSON.stringify(event),
+            value: Buffer.from(JSON.stringify(event)),
             timestamp,
-            offset: event.offset,
-            key: event.team_id + ':' + event.distinct_id,
+            offset: i,
+            key: withKey ? makeMessageKey(event) : null,
+            size: 0, // irrelevant, but needed for type checking
         }))
     }
 
-    beforeEach(() => {
+    beforeEach(async () => {
+        ;[hub, closeServer] = await createHub()
         queue = {
             bufferSleep: jest.fn(),
-            pluginsServer: {
-                INGESTION_CONCURRENCY: 4,
-                kafkaProducer: {
-                    produce: jest.fn(),
-                },
-                db: 'database',
-            },
+            pluginsServer: hub,
         }
         jest.mock('./../../../src/worker/ingestion/event-pipeline/runner')
     })
 
+    afterEach(async () => {
+        await closeServer()
+        jest.clearAllMocks()
+    })
+
     it('reroutes events with no key to OVERFLOW topic', async () => {
-        const batch = [
-            {
-                partition: 0,
-                topic: KAFKA_EVENTS_PLUGIN_INGESTION,
-                value: JSON.stringify(captureEndpointEvent1),
-                timestamp: captureEndpointEvent1['timestamp'],
-                offset: captureEndpointEvent1['offset'],
-                key: null,
-                token: 'ok',
-            },
-        ]
+        const now = Date.now()
+        const [message] = createBatchWithMultipleEvents(
+            [captureEndpointEvent1],
+            now,
+            false // act as if this message was intended to be routed to overflow by capture
+        )
 
         const consume = jest.spyOn(ConfiguredLimiter, 'consume').mockImplementation(() => false)
+        const produce = jest.spyOn(queue.pluginsServer.kafkaProducer, 'produce')
 
         const tokenBlockList = buildStringMatcher('another_token,more_token', false)
-        await eachBatchParallelIngestion(tokenBlockList, batch, queue, IngestionOverflowMode.Reroute)
+        await eachBatchParallelIngestion(tokenBlockList, [message], queue, IngestionOverflowMode.Reroute)
 
         expect(consume).not.toHaveBeenCalled()
         expect(captureIngestionWarning).not.toHaveBeenCalled()
-        expect(queue.pluginsServer.kafkaProducer.produce).toHaveBeenCalledWith({
+        expect(produce).toHaveBeenCalledWith({
             topic: KAFKA_EVENTS_PLUGIN_INGESTION_OVERFLOW,
-            value: JSON.stringify(captureEndpointEvent1),
-            timestamp: captureEndpointEvent1['timestamp'],
-            offset: captureEndpointEvent1['offset'],
```

Comment:
These were undefined so they weren't being considered in this assertion.

`timestamp` _can_ be provided to `produce`, if we want to preserve it across rerouting.